### ISSUE TITLE
RStudio fails with "The R Session had a fatal error" and "site can't be reached" after upgrade to 1.2

### ISSUE THREAD

**Issue by gtritchie** [rstudio/rstudio#2313]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.2.376-1
    OS Version      : macOS ????
    R Version       : ????

### Steps to reproduce the problem
Install RStudio 1.2. Launch it.

### Describe the problem in detail
Get the following:

<img width="456" alt="screen shot 2018-03-03 at 11 10 48 am" src="https://user-images.githubusercontent.com/10569626/36937351-b19dac3a-1ec6-11e8-9933-c4adbe603435.png">

<img width="689" alt="screen shot 2018-03-03 at 11 10 53 am" src="https://user-images.githubusercontent.com/10569626/36937356-be1bcbcc-1ec6-11e8-9ebf-5b7704cdd6f7.png">

### Describe the behavior you expected

RStudio should start.

**Comment by kevinushey** [comment]
Did you recently upgrade R as well, or only RStudio in this case? Our theory in the past is that this is caused by inadvertently mixing different R versions on launch.

**Comment by javierluraschi** [comment]
I hit this one back in February... the only change was upgrading to 1.2, reverting back to 1.1 made fixed this issue. I was able to use 1.2 by removing all R versions from `/Library/Frameworks/R.framework/Versions/` and reinstalling R, otherwise was hitting:

<img width="1023" alt="screen shot 2018-02-05 at 11 48 04 pm 1" src="https://user-images.githubusercontent.com/3478847/36937719-73fc65e6-1ecc-11e8-8962-623e68760fe2.png">

**Comment by kevinushey** [comment]
I _really_ want to understand how R / RStudio gets into this state as a number of users have encountered this issue over the past couple years, and "try deleting R and RStudio and then reinstalling" isn't really the best prescription.

I suspect this is the code that's failing to run:

https://github.com/rstudio/rstudio/blob/79c176e69a17b6b4270073524140ea7c95661179/src/cpp/r/session/RSession.cpp#L433-L437

In other words, our attempt to source the `Tools.R` file is failing.

Can we pause the `rsession` at the time when the crash occurs, and see what packages / libraries are loaded? Are we loading the same `libR.so` + associated R libraries that we expect to be loading?

**Comment by kevinushey** [comment]
I think I'm correct that the wrong `libR.dylib` is being loaded. Copying notes from Slack.
---
With RStudio v1.2:

```kevin@cdrv:/Applications/RStudio.app/Contents/MacOS
$ otool -L rsession
rsession:
        /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit (compatibility version 45.0.0, current version 1504.83.101)
        /Library/Frameworks/R.framework/Versions/3.0/Resources/lib/libR.dylib (compatibility version 3.0.0, current version 3.0.3)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.60.2)
        /usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.8)
        /System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices (compatibility version 1.0.0, current version 775.19.0)
        @executable_path/../Frameworks/libssl.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
        @executable_path/../Frameworks/libcrypto.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
        /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 307.5.0)
        /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1349.8.0)
        /System/Library/Frameworks/Foundation.framework/Versions/C/Foundation (compatibility version 300.0.0, current version 1349.81.0)
        /usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
```
Note that we're implicitly linking to R 3.0.x. If the user has R 3.0.x installed on their machine, I suspect RStudio will inadvertently load that version of R on launch rather than the one actually requested at runtime.
@hadley: do you have R 3.0.x installed? (edited)
If this is the reason why things are blowing up now, I suspect we got away with it in the past because we linked to a super-old version of R that nobody actually had installed on their machines

kevin [12:36 PM]
Okay, I can reproduce the crash. I think I'm right in my assessment. If you install R 3.0.3, then switch back to R 3.x.y, then you'll see that crash on launch.
I'll copy this to the issue.

**Comment by lingingrid** [comment]
I'm using Mac and I was also able to use 1.2 by removing all old versions from from /Library/Frameworks/R.framework/Versions/

I assume that the new version of Rstudio doesn't know which R to use when there're multiple paths exist.

**Comment by hadley** [comment]
Yes, I had 3.0 installed, and deleting it fixed the issue.

**Comment by ronblum** [comment]
Confirmed on Mac Desktop 1.2.405 with R 3.0.3 and 3.4.3 installed.

**Comment by Dzivo** [comment]
On windows rstudio doesnt work with 3.0.3 as well

**Comment by gtritchie** [comment]
This issue is 8 years old.

RStudio stopped supporting R  versions < 3.3.0 in the 2022.07 release. https://docs.posit.co/ide/news/#deprecated-removed-11
